internal/rest/resources/tokens.go

```diff
@@ -43,6 +45,11 @@ func tokensPost(state state.State, r *http.Request) response.Response {
 		return response.BadRequest(err)
 	}
 
+	err = utils.ValidateFQDN(req.Name)
+	if err != nil {
+		return response.SmartError(fmt.Errorf("Invalid token name %q: %w", req.Name, err))
```

Comment:
I think we should be explicit and report `Token name is not a valid FQDN: %w`

Reply:
Fixed. I have also updated the other messages where we return the error from `ValidateFQDN`.